web-ts/src/components/LocationHeader.tsx

```diff
@@ -1,47 +1,63 @@
-import React, { ReactNode } from 'react'
-import { withTranslation, TFunction } from 'react-i18next'
+import React, { ReactNode, ReactElement } from 'react'
+import { generatePath } from 'react-router-dom'
+import { useTranslation } from 'react-i18next'
 import LanguageSelector from './LanguageSelector'
 import searchIcon from '../assets/magnifier.svg'
 import landingIcon from '../assets/location-icon.svg'
 import Header from './Header'
 import HeaderNavigationItem from '../components/HeaderNavigationItem'
-import { CityModel } from 'api-client'
+import {
+  CATEGORIES_ROUTE,
+  CityModel,
+  EVENTS_ROUTE,
+  LANDING_ROUTE,
+  LOCAL_NEWS_TYPE,
+  OFFERS_ROUTE,
+  POIS_ROUTE,
+  SEARCH_ROUTE,
+  TU_NEWS_TYPE
+} from 'api-client'
 import offersIcon from '../assets/offers.svg'
 import localInformationIcon from '../assets/local_information.svg'
 import eventsIcon from '../assets/events.svg'
 import newsIcon from '../assets/news.svg'
 import poisIcon from '../assets/pois.svg'
 import HeaderActionBarItemLink from '../components/HeaderActionItemLink'
 import buildConfig from '../constants/buildConfig'
+import { RoutePatterns } from '../routes/RootSwitcher'
 
 type PropsType = {
   cityModel: CityModel
   pathname: string
   languageCode: string
   viewportSmall: boolean
-  t: TFunction
   onStickyTopChanged: (stickyTop: number) => void
   languageChangePaths: Array<{ code: string; path: string | null; name: string }> | null
 }
 
-export class LocationHeader extends React.Component<PropsType> {
-  getActionItems(): Array<ReactNode> {
-    const { languageCode, pathname, languageChangePaths, t } = this.props
+const LocationHeader = (props: PropsType): ReactElement => {
+  const { viewportSmall, onStickyTopChanged, cityModel, languageCode, pathname, languageChangePaths } = props
+  const { eventsEnabled, poisEnabled, offersEnabled, tunewsEnabled, pushNotificationsEnabled } = cityModel
 
+  const params = { cityCode: cityModel.code, languageCode }
+  const categoriesPath = generatePath(RoutePatterns[CATEGORIES_ROUTE], params)
+  const eventsPath = generatePath(RoutePatterns[EVENTS_ROUTE], params)
```

Comment:
Same as above it would be better if `generatePath` would do the lookup in `RoutePatterns` and handle errors there.

Reply:
`generatePath` is a function of `react-router`, see https://reactrouter.com/web/api/generatePath. We could introduce a custom wrapper though, I think I would like that :+1: Would do that in a separate PR though. We also need checks whether the current route is a specific route.

Reply:
Will do this in https://issues.tuerantuer.org/browse/IGAPP-668